modules_extra/network.js

```diff
@@ -100,77 +95,166 @@ function duration (s) {
-                  peer.stateChange && ('(' + human(new Date(peer.stateChange))) + ')')
-                ),
-                'source:'+peer.source,
-                h('pre', legacyToMultiServer(peer)),
-                h('button', 'connect', {onclick: function () {
-                  api.sbot_gossip_connect(peer, function (err) {
+    menu_items: () => h('a', {href: '#/network'}, '/network'),
+    builtin_tabs: () => ['/network'],
+    screen_view,
+    mcss: () => fs.readFileSync(__filename.replace(/js$/, 'mcss'), 'utf8')
+  }
+
+  function screen_view (path) {
+    if (path !== '/network') return
+
+    var peers = obs_gossip_peers(api)
+
+    return h('div', { style: {'overflow':'auto'}, className: 'column scroller' }, [
+      h('Network', [
+        mutantMap(peers, peer => {
+          var { key, ping, source, state, stateChange } = peer
+
+          return h('NetworkConnection', [
+            h('section.avatar', [
+              api.avatar_image_link(key()),
+            ]),
+            h('section.name', [
+              api.avatar_name_link(key()),
+            ]),
+            h('section.type', [
+              computed(peer, getType),
+            ]),
+            h('section.source', [
+              'source: ',
+              h('code', source)
+            ]),
+            h('section.state', [
+              'state: ',
+              h('i', {
+                className: computed(state, (state) => '-'+state)
+              }),
+              h('code', when(state, state, 'not connected'))
+            ]),
+            h('section.actions', [
+              h('button', {
+                'ev-click': () => {
+                  api.sbot_gossip_connect(peer(), (err) => {
                     if(err) console.error(err)
-                    else console.log('connected to', peer)
+                    else console.log('connected to', peer())
                   })
-                }})
+                }},
+                'connect'
               )
-            )
-          })
-
-          setTimeout(poll, 5000)
+            ]),
+            h('section.time-ago', [
+              h('div',
+                { title: computed(stateChange, formatDate) },
+                [ computed(stateChange, humanDate) ]
+              )
+            ]),
+            h('section.ping', [
+              h('div.rtt', [
+                'rtt: ',
+                h('code', computed(ping.rtt.mean, duration))
+              ]),
+              h('div.skew', [
+                'skew: ',
+                h('code', computed(ping.skew.mean, duration))
+              ]),
+            ]),
+            h('section.address', [
+              h('code', computed(peer, legacyToMultiServer))
+            ])
+          ])
         })
+      ])
+    ])
+  }
+}
 
-      })()
-
-      return h('div.column.scroll-y', ol)
-    }
+function obs_gossip_peers (api) {
+  var timer = null
+  var state = Dict({}, {
+    //
+    // TODO - get this api implemented in mutantDict
```

Comment:
this is implemented now (https://github.com/mmckegg/mutant/issues/4), can we clean this up?